blockchain/chain_sync/src/sync.rs

```diff
@@ -514,12 +516,34 @@ where
         Ok(fts)
     }
     // Block message validation checks
-    fn check_blk_msgs(&self, block: Block, _tip: &Tipset) -> Result<(), Error> {
-        // TODO retrieve bls public keys for verify_bls_aggregate
-        // for _m in block.bls_msgs() {
-        // }
-        // TODO verify_bls_aggregate
-
+    fn check_block_msgs(&self, block: Block, tip: &Tipset) -> Result<(), Error> {
+        let mut pub_keys = Vec::new();
+        let mut cids = Vec::new();
+        for m in block.bls_msgs() {
+            let pk = self
+                .state_manager
+                .get_bls_public_key(m.from(), tip.parent_state())?;
+            pub_keys.push(pk);
+            cids.push(m.cid()?.to_bytes());
+        }
+        if let Some(sig) = block.header().bls_aggregate() {
```

Comment:
Should probably error if there is no bls aggregate signature on the block, right?